setup.py

```diff
@@ -36,25 +36,39 @@
         'Programming Language :: Python :: 3.4',
         'Topic :: Scientific/Engineering',
     ],
+    # If you add/remove a requirement, please also update full
+    extras_require = {
+        'pandas': ['numpy', 'pandas', 'scipy', 'statsmodels'],
+        'graphics': ['matplotlib', 'mpld3', 'seaborn'],
+        'lomb_scargle': ['astroML', 'supersmoother', 'gatspy'],
+        'full': [
+            'numpy', 'pandas', 'scipy', 'statsmodels',
+            'matplotlib', 'mpld3', 'seaborn',
+            'astroML', 'supersmoother', 'gatspy',
+        ],
+    },
     entry_points={
         'console_scripts': [
-            'p.cdf = pandashells.bin.p_cdf:main',
             'p.config = pandashells.bin.p_config:main',
             'p.crypt = pandashells.bin.p_crypt:main',
-            'p.df = pandashells.bin.p_df:main',
-            'p.facet_grid = pandashells.bin.p_facet_grid:main',
             'p.format = pandashells.bin.p_format:main',
-            'p.hist = pandashells.bin.p_hist:main',
-            'p.linspace = pandashells.bin.p_linspace:main',
-            'p.lomb_scargle = pandashells.bin.p_lomb_scargle:main',
-            'p.merge = pandashells.bin.p_merge:main',
             'p.parallel = pandashells.bin.p_parallel:main',
-            'p.plot = pandashells.bin.p_plot:main',
-            'p.rand = pandashells.bin.p_rand:main',
-            'p.regplot = pandashells.bin.p_regplot:main',
-            'p.regress = pandashells.bin.p_regress:main',
-            'p.example_data = pandashells.bin.p_example_data:main',
-            'p.sig_edit = pandashells.bin.p_sig_edit:main',
+
+            'p.df = pandashells.bin.p_df:main [pandas]',
```

Comment:
I see.  Hmmm... I think I might actually not want this behavior as I have some custom import logic checking built into Pandashells.  If a tool is missing a requirement, I have a nice error message that pops up informing the user of the steps needed to install that package.

On the other hand, I might consider removing that logic from the code base and use this strategy (removing code is almost always a good thing, in my opinion), but I think I have bigger fish to fry in the package right now.  Thanks for the info.